src/lib/SHM.cpp

```diff
@@ -193,8 +194,15 @@ SharedTexture::SharedTexture(
   r->Populate(d3d, header.sequenceNumber);
   auto key = GetTextureKeyFromSequenceNumber(header.sequenceNumber);
   if (r->mMutex->AcquireSync(key, 10) != S_OK) {
-    mResources = nullptr;
-    return;
+    dprintf("Failed to get mutex with key {}, attempting repopulating", key);
+    // try repopulating (e.g. if Openkneeboard process has been restarted):
+    r->mTexture = nullptr;
+    r->Populate(d3d, header.sequenceNumber);
+    if (r->mMutex->AcquireSync(key, 10) != S_OK) {
+      dprintf("Failed to get mutex with key {}, after repopulating", key);
+      mResources = nullptr;
+      return;
+    }
```

Comment:
Ah, prior to the introduction of opacity, the resources were cast/checked locally

Can you try:

* replace `r->mTexture = nullptr` with `*r = {}` to get every field back to a known-good initial state
* revert to previous behavior - just clearing mResources - if the return value is not `WAIT_ABANDONED` - in particular, we hopefully don't need to tear down and recreate everything in case of `WAIT_TIMEOUT`, which is occasionally expected